python/lsst/analysis/tools/actions/plot/gridPlot.py

```diff
@@ -90,16 +103,48 @@ class GridPlot(PlotAction):
     def __call__(self, data: KeyedData, **kwargs) -> PlotResultType:
         """Plot data."""
         fig = plt.figure(figsize=self.figsize, dpi=self.dpi)
+
+        if self.height_ratios is None:
+            height_ratios = np.ones(self.numRows) / self.numRows
+        else:
+            height_ratios = self.height_ratios / np.sum(self.height_ratios)
+
+        if self.width_ratios is None:
+            width_ratios = np.ones(self.numCols) / self.numCols
+        else:
+            width_ratios = self.width_ratios / np.sum(self.width_ratios)
+
+        includesPlotInfoPanel = False
+        addPlotInfoRow = 1
+        for panel in self.panels.values():
+            if isinstance(panel.plotElement, PlotInfoElement):
+                includesPlotInfoPanel = True
+                addPlotInfoRow = 0
+                break
+
+        if not includesPlotInfoPanel:
+            height_ratios = np.append([1.0], 20.0 * height_ratios)
+
         if self.suptitle is not None:
             fig.suptitle(**self.suptitle)
-        gs = GridSpec(self.numRows, self.numCols, figure=fig)
+        gs = GridSpec(
+            self.numRows + addPlotInfoRow,
+            self.numCols,
+            figure=fig,
+            height_ratios=height_ratios,
+            width_ratios=width_ratios,
+        )
+
+        if not includesPlotInfoPanel:
+            ax = fig.add_subplot(gs[0, :])
+            _ = PlotInfoElement().makeElement(ax=ax, fig=fig, **kwargs)
```

Comment:
Ok. It's used to calculate a scale-independent spacing between the plotInfo title and the remaining text. I can see if there's another way to do it. What's the justification for not wanting specific elements (as opposed to all) being passed a fig object?

Reply:
I think it is a violation of separation of concerns and encapsulation. I think it will also encourage others to look at this, also pass the figure object, and have the elements make changes to the figure. If all you need in the dpi spacing or something like that either pass that, or possibly look to see if that is context we think other elements may need to know.

Reply:
Thanks - makes total sense. I'll pass the fig scale transform instead, which is all that it's needed for.